client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/index.jsx

```diff
@@ -122,7 +123,27 @@ class VisibleSubmissionEditIndex extends Component {
     } = this.props;
     const published = workflowState === workflowStates.Published;
     dispatch(
-      saveGrade(params.submissionId, Object.values(grading), exp, published),
+      saveAllGrade(params.submissionId, Object.values(grading), exp, published),
+    );
+  }
+
+  handleSaveGrade(id) {
+    const {
+      dispatch,
+      match: { params },
+      grading,
+      exp,
+      questions,
+    } = this.props;
+    dispatch(
+      saveGrade(
+        params.submissionId,
+        [grading[id]],
+        Object.values(grading),
+        exp,
+        false,
```

Comment:
How do you know this is always `false`?

Reply:
it's not that I know it will be always false, but rather that it needs to be false.
this function will only be called when we want to update the grade for one question and not the entire submission, and hence we should never allow the experience points to be awarded as points, but rather as draft.

Only when the grader decided to publish the grade (by clicking on the button "Publish Grade" on the bottom of the page) then the points awarded will be awarded as points

Reply:
What if the submission is already in published mode and the exp is updated?

Reply:
okay there are 2 possibilities this might happen, on how exp is updated:
1. exp is updated directly at the EXP Points box inside the submission. In this case, this function will contribute nothing to it, as it will never be invoked in the first place
2. it's updated because we modify the grade in one of the question. This one, to be honest, I was not aware that the grade can be updated even after it's been published. My apologies for that I'm unaware about this situation, and I have updated it accordingly. And you are correct that published might be True.